### ISSUE TITLE
flux_bootstrap_git better commits

### ISSUE THREAD

**Issue by sereinity** [fluxcd/terraform-provider-flux#550]
Hello, I want to switch to the `flux_bootstrap_git` resource. But currently, it breaks our CI as we enforce that our commits match the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0]) norm.

I suggest two way to perform that:
* add a `commit_message_prefix` parameter
* each commit message may be overridden by the provider settings (I found only 3 different commit messages in the code).

**Comment by swade1987** [comment]
Hello @anupclouddevops  ,

I hope you're doing well! I'm the newest contributor to this repository, and I'm currently in the process of issue grooming to ensure that all concerns are addressed promptly and efficiently.

I noticed this issue you reported and wanted to check in with you to see if it's still affecting your work. Your feedback is invaluable to us, and any additional insights or updates you can share would be greatly appreciated to help us understand and solve the problem more effectively.

If this issue has been resolved, could you please share how it was fixed? This information could be incredibly helpful to others in the community facing similar problems. It would also allow us to close this issue with a clear resolution.
In case the issue is still open and troubling you, let's work together to find a solution. Your satisfaction and the smooth functioning of our project are our top priorities.

Thank you for your time and contributions to our community. Looking forward to your response!

Best regards,

Steve

-----

You can leverage `commit_message_appendix` [here](https://registry.terraform.io/providers/fluxcd/flux/latest/docs#commit_message_appendix) in the `git` block on the flux provider configuration to resolve this.

**Comment by sereinity** [comment]
Please reopen, I had read the doc and already known about `commit_message_appendix` at the time. It doesn't fit my needs.


As I said, the CI check that commits added to the repository matches [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), adding an appendix doesn't help.

Example of wanted commits:
- `Add Flux sync manifests` :arrow_right: `feat(myenvname): add Flux sync manifests`
- `Add Flux v0.1.2 component manifests` :arrow_right: `feat(myenvname): add Flux v0.1.2 component manifests`
- `Update Flux` :arrow_right: `fix(myenvname): upgrade Flux to v1.2.3`

Or even use `chore` type, or whatever the user want the commit type to be on upgrading or bootstrapping flux (may be different commit types).

As my examples show it, the scope should also be settable by the user.

**Comment by swade1987** [comment]
@stefanprodan is my understanding correct that this would need to be supported in flux itself first before we can allow this functionality with the provider.

**Comment by swade1987** [comment]
@sereinity, I propose you add an enhancement request to the `fluxcd/flux2` repository, and then I will close this one out. If/when the enhancement is implemented, we can add the supporting functionality to the provider.

**Comment by sereinity** [comment]
Closing in favor of a change upstream, like the one requested in: https://github.com/fluxcd/flux2/issues/2082
